packages/openapi3/src/openapi.ts

```diff
@@ -1464,7 +1464,12 @@ function serializeDocument(root: OpenAPI3Document, fileType: FileType): string {
     case "json":
       return prettierOutput(JSON.stringify(root, null, 2));
     case "yaml":
-      return yaml.dump(root, { noRefs: true });
+      return yaml.dump(root, {
+        noRefs: true,
+        replacer: function (key, value) {
+          return key === "$ref" && value instanceof Ref ? value.value! : value;
```

Comment:
I think this would fix the discriminator mapping issue by not checking key, and preserve the assert in Ref.toJSON() when emitting Yaml.

```suggestion
          return value instanceof Ref ? value.toJSON() : value;
```

Reply:
Thanks!

Reply:
Hmm, I'm still seeing it on playground, which "try it" claims to have updated, checking

Reply:
"Unresolving" only to show conversation.

Hmm, but the need to rebaseline looks promising, maybe the playground update didn't work for some reason.

![image](https://user-images.githubusercontent.com/75470/212967417-57dbaa35-0ed2-47f7-98ac-e5871276a6a3.png)

Reply:
Nevermind it's working now, some sort of delay from time it is pushed to playground and time it is reflected in browser, I guess.

Reply:
Yeah, I've noticed that before too